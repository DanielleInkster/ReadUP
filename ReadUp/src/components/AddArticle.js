import React, {useState} from 'react';
import {
  View,
  Text,
  TextInput,
  TouchableOpacity,
  StyleSheet,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

Icon.loadFont();

const AddArticle = ({getData}) => {
  const [articleText, setArticleText] = useState('');

  const onChange = (textValue) => setArticleText(textValue);

  return (
    <View>
      <TextInput
        value={articleText}
        testID="inputText"
        placeholder="Enter URL"
        style={styles.input}
        onChangeText={onChange}
      />

      <View style={{flexDirection: 'row'}}>
        <TouchableOpacity
          testID="addButton"
          style={styles.btn}
          onPress={() => getData(articleText)}>
          <Text style={styles.btnText}>
            <Icon name="plus" size={15} />
            {'  Add Article'}
          </Text>
        </TouchableOpacity>

        <TouchableOpacity
          testID="clearButton"
          style={styles.btn}
          onPress={() => setArticleText('')}>
          <Text style={styles.btnText}>
            <Icon name="remove" size={15} />
            {'  Clear'}
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  input: {
    backgroundColor: '#bbe1fa',
    height: 70,
    padding: 8,
    borderRadius: 15,
    marginTop: '5%',
    marginHorizontal: 5,
    fontSize: 20,
    borderColor: '#1b262c',
    borderStyle: 'solid',
    borderWidth: 1.5,
  },

  btn: {
    backgroundColor: '#0f4c75',
    borderRadius: 15,
    padding: 9,
    height: 50,
    width: '40%',
    marginVertical: '6%',
    marginHorizontal: '5%',
    justifyContent: 'center',
    alignItems: 'center',
  },

  btnText: {
    color: '#bbe1fa',
    fontSize: 20,
    textAlign: 'center',
  },
});

export default AddArticle;
