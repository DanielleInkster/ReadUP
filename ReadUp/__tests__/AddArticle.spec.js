import React from 'react';
import {render, fireEvent} from 'react-native-testing-library';
import AddArticle from '../src/components/AddArticle';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<AddArticle />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('AddArticle', () => {
  describe('clicking clear', () => {
    it('clears the input field', () => {
      const {getByTestId} = render(<AddArticle />);

      fireEvent.changeText(getByTestId('inputText'), 'Hello world');
      fireEvent.press(getByTestId('clearButton'));
      expect(getByTestId('inputText').props.value).toEqual('');
    });
  });
});
