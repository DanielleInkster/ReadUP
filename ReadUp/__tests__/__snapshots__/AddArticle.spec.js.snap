// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Enter URL"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#bbe1fa",
        "borderColor": "#1b262c",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1.5,
        "fontFamily": "sans-serif-medium",
        "fontSize": 20,
        "height": 70,
        "marginHorizontal": 5,
        "padding": 8,
      }
    }
    testID="inputText"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0f4c75",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "marginHorizontal": "5%",
          "marginVertical": "6%",
          "opacity": 1,
          "padding": 9,
          "width": "40%",
        }
      }
      testID="addButton"
    >
      <Text
        style={
          Object {
            "color": "#bbe1fa",
            "fontFamily": "sans-serif-medium",
            "fontSize": 20,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
          Add Article
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0f4c75",
          "borderRadius": 10,
          "height": 50,
          "justifyContent": "center",
          "marginHorizontal": "5%",
          "marginVertical": "6%",
          "opacity": 1,
          "padding": 9,
          "width": "40%",
        }
      }
      testID="clearButton"
    >
      <Text
        style={
          Object {
            "color": "#bbe1fa",
            "fontFamily": "sans-serif-medium",
            "fontSize": 20,
            "fontWeight": "600",
            "textAlign": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 15,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
          Clear
      </Text>
    </View>
  </View>
</View>
`;
